stacker/src/arena_hashmap.rs

```diff
@@ -284,31 +304,36 @@ impl ArenaHashMap {
         }
         let hash = self.get_hash(key);
         let mut probe = self.probe(hash);
+        let mut bucket = probe.next_probe();
```

Comment:
what is the goal of putting these outside of the loop?

Reply:
To fetch the next bucket before the loop jmp. The comment for it is at the end of the loop.
The idea behind it is to allow unconditionally prefetching data (without the branch predictor)

Reply:
What do you mean by unconditionally:  this is not a conditional jump here.
also... 
I would not be surprised if for LLVM, the two are exactly the same...

Reply:
Sorry, that part with the branch predictor didn't make sense. It helps the instruction prefetcher a little bit I think.
The two version are indeed the same, but only with unsafe access. With bounds checks llvm can't optimize the code as well.

https://godbolt.org/z/x7xc8sTcY